CalibTracker/SiStripChannelGain/src/APVGainHelpers.cc

```diff
@@ -0,0 +1,194 @@
+ * tell in which side the wheel is sitting.
+ */
+int APVGain::subdetectorPlane(uint32_t det_id, const TrackerTopology* topo) {
+  if( topo ) {
+    if( APVGain::subdetectorId(det_id)==StripSubdetector::TIB )      return topo->tibLayer(det_id);
+    else if( APVGain::subdetectorId(det_id)==StripSubdetector::TID ) return (2*topo->tidSide(det_id)-3)*topo->tidWheel(det_id);
+    else if( APVGain::subdetectorId(det_id)==StripSubdetector::TOB ) return topo->tobLayer(det_id);
+    else if( APVGain::subdetectorId(det_id)==StripSubdetector::TEC ) return (2*topo->tecSide(det_id)-3)*topo->tecWheel(det_id);
+  }
+  return 0;
+};
+
+/** Brief Extract from a char * the subdetector type.
+ * Return an integer whioch is the detector plane where the module sits.
+ * The char * string is expected to have the subdetector plane put at its
+ * end after an "_" char.
+ */
+int APVGain::subdetectorPlane(const char* tag) {
+   std::string s = std::string(tag);
+   std::size_t p = (s.find("layer")!=std::string::npos)? s.find("layer") : s.find("wheel");
+   if (p!=std::string::npos) {
+       std::size_t start = s.find("_",p+1) + 1;
+       std::size_t stop  = s.find('_',start);
+       std::string plane = s.substr(start,stop-start);
+       return atoi( plane.c_str());
+   }
+   return 0;
+};
+
+
+/** Brief Fetch the Monitor Element corresponding to a DetId.
+ *  */
+std::vector<MonitorElement*> APVGain::FetchMonitor(std::vector<APVGain::APVmon> histos, uint32_t det_id, 
+                                                                                        const TrackerTopology* topo) {
+    std::vector<MonitorElement*> found = std::vector<MonitorElement*>();
+    int sId    = APVGain::subdetectorId((uint32_t)det_id);
+    int sPlane = APVGain::subdetectorPlane((uint32_t)det_id, topo);
+    int sSide  = APVGain::subdetectorSide((uint32_t)det_id, topo);
+    std::vector<APVGain::APVmon>::iterator it= histos.begin();
+    
+    while (it!=histos.end()) {
+        //std::string tag = (*it)->getName();
+        int subdetectorId    = (*it).subdetectorId;
+        int subdetectorSide  = (*it).subdetectorSide;
+        int subdetectorPlane = (*it).subdetectorPlane;
+
+        bool match = (subdetectorId==0    || subdetectorId==sId) &&
+                     (subdetectorPlane==0 || subdetectorPlane==sPlane) &&
+                     (subdetectorSide==0  || subdetectorSide==sSide);
+
+        if (match) {
+            found.push_back((*it).monitor);
+        }
+        it++;
+    }
+    return found;
+}
+
+
+
+std::vector<std::pair<std::string,std::string>> 
+APVGain::monHnames(std::vector<std::string> VH, bool allPlanes, const char* tag) {
+
+    std::vector<std::pair<std::string,std::string>> out;
+    std::string Tag = tag;
+    if (Tag.length())  Tag = "__" + Tag;
+
+    std::string h_tag = "";
+    std::string h_tit = "";
+
+    if (allPlanes) {
+        // Names of monitoring histogram for TIB layers
+        for(unsigned int i=1;i<5;i++) {
```

Comment:
please document the magic numbers and add a out.reserve(40) [or whatever all the loops add up to]